src/componentize.js

```diff
@@ -89,89 +82,105 @@ export async function componentize(opts,
         );
       }
       opts = _deprecatedWitWorldOrOpts;
     }
   }
 
-  const tmpDir = join(
-    tmpdir(),
-    createHash('sha256')
-      .update(Math.random().toString())
-      .digest('hex')
-      .slice(0, 12)
-  );
-  await mkdir(tmpDir);
-  const sourceDir = maybeWindowsPath(join(tmpDir, 'sources'));
-  await mkdir(sourceDir);
+  // Prepare a working diretory for use during componentization
+  const { sourcesDir, baseDir: workDir } = await prepWorkDir();
 
   let {
     sourceName = 'source.js',
-    sourcePath = join(sourceDir, sourceName),
+    sourcePath = maybeWindowsPath(join(sourcesDir, sourceName)),
     preview2Adapter = preview1AdapterReactorPath(),
     witPath,
     witWorld,
     worldName,
     disableFeatures = [],
     enableFeatures = [],
+
+    debug = { ...DEFAULT_DEBUG_SETTINGS },
     debugBuild = false,
-    runtimeArgs,
     debugBindings = false,
     enableWizerLogging = false,
-    aotCache = fileURLToPath(
-      new URL(`../lib/starlingmonkey_ics.wevalcache`, import.meta.url),
-    ),
+
+    runtimeArgs,
+
+    aotCache = DEFAULT_AOT_CACHE,
   } = opts;
 
-  const engine =
-    opts.engine ||
-    fileURLToPath(
-      new URL(
-        opts.enableAot
-          ? `../lib/starlingmonkey_embedding_weval.wasm`
-          : `../lib/starlingmonkey_embedding${debugBuild ? '.debug' : ''}.wasm`,
-        import.meta.url,
-      ),
-    );
+  debugBindings = debugBindings || debug?.bindings;
+  debugBuild = debugBuild || debug?.build;
+  enableWizerLogging = enableWizerLogging || debug?.enableWizerLogging;
+
+  // Determine the path to the StarlingMonkey binary
+  const engine = getEnginePath(opts);
 
   let { wasm, jsBindings, exports, imports } = spliceBindings(
     await readFile(engine),
     witWorld,
     maybeWindowsPath(witPath),
     worldName,
-    false
+    false,
   );
 
-  const input = join(tmpDir, 'in.wasm');
-  const output = join(tmpDir, 'out.wasm');
+  const inputWasmPath = join(workDir, 'in.wasm');
+  const outputWasmPath = join(workDir, 'out.wasm');
 
-  await writeFile(input, Buffer.from(wasm));
-  await writeFile(join(sourceDir, "initializer.js"), jsBindings);
+  await writeFile(inputWasmPath, Buffer.from(wasm));
+  let initializerPath = maybeWindowsPath(join(sourcesDir, 'initializer.js'));
+  await writeFile(initializerPath, jsBindings);
 
   if (debugBindings) {
-    console.log('--- JS Bindings ---');
-    console.log(
-      jsBindings
-        .split('\n')
-        .map((ln, idx) => `${(idx + 1).toString().padStart(4, ' ')} | ${ln}`)
-        .join('\n')
-    );
-    console.log('--- JS Imports ---');
-    console.log(imports);
-    console.log('--- JS Exports ---');
-    console.log(exports);
+    // If a bindings output directory was specified, output generated bindings to files
+    if (debug?.bindingsDir) {
+      console.error(`outputting debug files to [${debug?.bindingsDir}]...\n`);
```

Comment:
"Outputting" sounds a bit jarring to me :)
```suggestion
      console.error(`Storing debug files in "${debug?.bindingsDir}"\n`);
```